dotCMS/src/main/java/com/dotcms/publisher/util/DependencyManager.java

```diff
@@ -621,119 +688,155 @@ private void setHTMLPagesDependencies(Set<String> idsToWork) {
 				Identifier iden = idenAPI.find(pageId);
 
 				// Host dependency
-				Host h = APILocator.getHostAPI().find(iden.getHostId(), user, false);
-				hosts.addOrClean( iden.getHostId(), h.getModDate());
-				hostsSet.add(iden.getHostId());
-				Folder folder = folderAPI.findFolderByPath(iden.getParentPath(), iden.getHostId(), user, false);
-				folders.addOrClean( folder.getInode(), folder.getModDate());
-				foldersSet.add(folder.getInode());
+				if (publisherFilter.acceptExcludeDependencyClasses(PusheableAsset.SITE.getType())) {
+					Host h = APILocator.getHostAPI().find(iden.getHostId(), user, false);
+					hosts.addOrClean(iden.getHostId(), h.getModDate());
+					hostsSet.add(iden.getHostId());
+				}
 
+				// Folder dependencies
+				if (publisherFilter.acceptExcludeDependencyClasses(PusheableAsset.FOLDER.getType())) {
+					Folder folder = folderAPI
+							.findFolderByPath(iden.getParentPath(), iden.getHostId(), user, false);
+					folders.addOrClean(folder.getInode(), folder.getModDate());
+					foldersSet.add(folder.getInode());
+				}
 
 				// looking for working version (must exists)
 				IHTMLPage workingPage = null;
 
 				Contentlet contentlet = null;
-				try{
-					contentlet = APILocator.getContentletAPI().search("+identifier:"+pageId+" +working:true", 0, 0, "moddate", user, false).get(0);
+				try {
+					contentlet = APILocator.getContentletAPI()
+							.search("+identifier:" + pageId + " +working:true", 0, 0, "moddate",
+									user, false).get(0);
 				} catch (DotContentletStateException e) {
 					// content not found message is already displayed on console
-					Logger.debug(this, e.getMessage(),e);
+					Logger.debug(this, e.getMessage(), e);
```

Comment:
Logger should use the supplier such as
Logger.debug (this, ()->e.getMessage(), e)